### PR TITLE
Swtpg consolidation developments

### LINKED ISSUE #100: SWTPG consolidation developments
After the February coldbox, the SWTPG was updated to better monitor and handle different scenarios. Here is a summary of the main issues encountered:

- Need for configuration parameter for SWTPG threshold
- Need for configuration parameter for the SWTPG algorithm selection (threshold-based SWTPG or AbsRS)
- Need to add tp rate per channel to opmon

### PR DESCRIPTION
**Pull request by adam-abed-abud**
Resolves #100

### CODE REVIEW

**Review by gcrone — APPROVED** [review]
Looks OK to me

File: include/readoutlibs/models/detail/DefaultRequestHandlerModel.hxx [hunk 1]
```diff
@@ -380,6 +380,8 @@ DefaultRequestHandlerModel<RDT, LBT>::check_waiting_requests()
           m_waiting_requests.pop_back();
           size--;
         } else if (std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now() - m_waiting_requests[i].start_time).count() >= m_request_timeout_ms) {
+          uint64_t delay = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now() - m_waiting_requests[i].start_time).count();
+          TLOG() << "Charlie: The delay on this FAILED readout requests is: " << delay << " and the request timeout is: " << m_request_timeout_ms;
```
Comment (roland-sipos) [inline include/readoutlibs/models/detail/DefaultRequestHandlerModel.hxx]: Please ommit using of first names in debug messages.... 
Also, what does this delay calulcates?? If this is an important metric, it should be added to opmon, not as a log.

File: include/readoutlibs/models/detail/ReadoutModel.hxx [hunk 2]
```diff
@@ -333,7 +333,9 @@ ReadoutModel<RDT, RHT, LBT, RPT>::dispatch_requests(dfmessages::DataRequest& dat
      ers::error(RequestSourceIDMismatch(ERS_HERE, m_sourceid, data_request.request_information.component));
      return;
   }
-  TLOG_DEBUG(TLVL_QUEUE_POP) << "Received DataRequest" 
+
+  TLOG() << "Received DataRequest"
+  //TLOG_DEBUG(TLVL_QUEUE_POP) << "Received DataRequest"
```
Comment (roland-sipos) [inline include/readoutlibs/models/detail/ReadoutModel.hxx]: Spurious TLOG comment?

**Review by roland-sipos — CHANGES_REQUESTED** [review]
Remove possibly accidental LOG messages.

File: include/readoutlibs/models/detail/ReadoutModel.hxx (hunk 2 above)
Comment (roland-sipos) [reply include/readoutlibs/models/detail/ReadoutModel.hxx]: Also, this will spam logs!

**Comment by roland-sipos** [comment]
@glehmannmiotto can you please have a look on this? It has clashing changes with the request handling functiond.

**Comment by roland-sipos** [comment]
I would propose to close this PR, and open a fresh one just with the changes on the schema for the config and the info.

**Comment by roland-sipos** [comment]
Closing PR, as schema changes went in with another one.

**Pull request closed without merging**